localstack/services/events/event_ruler.py

```diff
@@ -0,0 +1,57 @@
+import logging
+import os
+from functools import cache
+from pathlib import Path
+
+from localstack import config
+from localstack.services.events.packages import event_ruler_package
+from localstack.services.events.utils import InvalidEventPatternException
+
+THIS_FOLDER = os.path.dirname(os.path.realpath(__file__))
+
+LOG = logging.getLogger(__name__)
+
+if config.EVENT_RULE_ENGINE == "java":
+    import jpype
+    import jpype.imports
+    from jpype import config as jpype_config
+    from jpype import java
+
+    # Workaround to unblock LocalStack shutdown. By default, JPype wait until all daemon threads are terminated,
+    # which blocks the LocalStack shutdown upon any leaked thread (know LS issue).
+    # See https://github.com/MPh-py/MPh/issues/15#issuecomment-778486669
+    jpype_config.destroy_jvm = False
```

Comment:
That's unfortunately needed. Could this cause any side-effects?

Reply:
The container shutdown will probably not work, as we have an orphan process which did not get terminated by its parent on shutdown. The supervisor will probably wait forever, until docker kills the container.